packages/react-native-web/src/modules/applyLayout/index.js

```diff
@@ -47,46 +101,49 @@ const applyLayout = Component => {
     function componentDidMount() {
       this._layoutState = emptyObject;
       this._isMounted = true;
-      this._onLayoutId = guid();
-      registry[this._onLayoutId] = this;
-      this._handleLayout();
+      if (this.props.onLayout) {
+        observe(this);
+      }
     }
   );
 
   Component.prototype.componentDidUpdate = safeOverride(
     componentDidUpdate,
-    function componentDidUpdate() {
-      this._handleLayout();
```

Comment:
Actually removing this might break backwards compatibility. Happy to fix this if you think it is necessary.

Reply:
fixed